ibstore/_store.py

```diff
@@ -454,25 +522,47 @@ def optimize_mm(
             chunksize=chunksize,
         )
 
-        for result in _minimized_blob:
-            inchi_key = result.inchi_key
-            molecule_id = self.get_molecule_id_by_inchi_key(inchi_key)
-            self.store_conformer(
-                MMConformerRecord(
+        inchi_to_id = {
+            inchi_key: id
+            for (id, inchi_key) in reversed(
+                db.db.query(
+                    DBMoleculeRecord.id,
+                    DBMoleculeRecord.inchi_key,
+                ).all()
+            )
+        }
+
+        with self._get_session() as db:
+            # from _mm_conformer_already_exists
+            seen = set(
+                db.db.query(
+                    DBMMConformerRecord.qcarchive_id,
+                )
+            )
```

Comment:
Should this not be filtered by the (this) force field? This looks like it fetches all QCArchive IDs of any record in the entire database.

If you minimized with force field A, this would populate this set with all of those results and then erroneously skip over them if minimization is later called on force field B. I don't think this is covered in unit tests since, but if my mental model is correct this should fail (pseudocode)

```python
test_partially_minimized():
    def get_n_results(store, ff1, ff2) -> tuple[int, ...]:
       return(
            len(store.get_all_mm_conformers(force_field=ff1),
            len(store.get_all_mm_conformers(force_field=ff2),
    )

    assert get_n_results(store, ffa, ffb) == (0, 0)

    store.optimize_mm(force_field=ffa)

    assert get_n_results(store, ffa, ffb) == (100, 0)  # say there are 100 total conformers

    store.optimize_mm(force_field=ffb)

    assert get_n_results(store, ffa, ffb) == (100, 100)  # it should fail here, returning (100, 0)
```

Reply:
Great point! I usually run one force field at a time, so I must have ignored that part of `_mm_conformer_already_exists`.

Reply:
7577463b1eff6023b35002f5c07a05c989c251dc